vertical-pod-autoscaler/recommender/model/vpa.go

```diff
@@ -70,53 +71,72 @@ type Vpa struct {
 	Conditions vpaConditionsMap
 	// Most recently computed recommendation. Can be nil.
 	Recommendation *vpa_types.RecommendedPodResources
-	// Pods controlled by this VPA object.
-	Pods map[PodID]*PodState
+	// All container aggregations that contribute to this VPA.
+	// TODO: Garbage collect old AggregateContainerStates.
+	aggregateContainerStates aggregateContainerStatesMap
 	// Value of the Status.LastUpdateTime fetched from the VPA API object.
 	LastUpdateTime time.Time
 	// Initial checkpoints of AggregateContainerStates for containers.
-	ContainerCheckpoints map[string]*AggregateContainerState
+	// The key is container name.
+	containerCheckpoints ContainerNameToAggregateStateMap
 }
 
 // NewVpa returns a new Vpa with a given ID and pod selector. Doesn't set the
-// links to the matched pods.
+// links to the matched aggregations.
 func NewVpa(id VpaID, selector labels.Selector) *Vpa {
 	vpa := &Vpa{
-		ID:                   id,
-		PodSelector:          selector,
-		Pods:                 make(map[PodID]*PodState), // Empty pods map.
-		ContainerCheckpoints: make(map[string]*AggregateContainerState),
+		ID:                       id,
+		PodSelector:              selector,
+		aggregateContainerStates: make(aggregateContainerStatesMap),
+		containerCheckpoints:     make(ContainerNameToAggregateStateMap),
 	}
 	return vpa
 }
 
-// MatchesPod returns true iff a given pod is matched by the Vpa pod selector.
-func (vpa *Vpa) MatchesPod(pod *PodState) bool {
-	if vpa.ID.Namespace != pod.ID.Namespace {
-		return false
+// UseAggregationIfMatching checks if the given aggregation matches (contributes to) this VPA
+// and adds it to the set of VPA's aggregations if that is the case.
+func (vpa *Vpa) UseAggregationIfMatching(aggregationKey AggregateStateKey, aggregation *AggregateContainerState) {
```

Comment:
Nit: In this convenience it should be named UseAggregationIfFirstAndMatching. Maybe TryUseAggregation? TryBindAggregation?